src/Controls/src/Core/BindingExpression.cs

```diff
@@ -460,7 +460,7 @@ internal static bool TryConvert(ref object value, BindableProperty targetPropert
 				var stringValue = value as string ?? string.Empty;
 				// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
 				// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
-				if (stringValue.EndsWith(".", StringComparison.Ordinal) && DecimalTypes.Contains(convertTo))
+				if (stringValue.EndsWith(CultureInfo.CurrentUICulture.NumberFormat.NumberDecimalSeparator, StringComparison.Ordinal) && DecimalTypes.Contains(convertTo))
```

Comment:
I think you want CurrentCulture and not CurrentUICulture here? CurrentCulture is more about input/output, and CurrentUICulture is for the text of the app (what language the dialog labels are in). Does that make sense? Or maybe I have it backwards?

Reply:
I _think_ that's correct @Eilon, at least for windows...

Maybe we should dig a but further to see what these map to on Android and iOS specifically...

@steveisok or @akoeplinger any guidance here?

Reply:
Yes `CurrentCulture` is the correct one for this case, not `CurrentUICulture`.

Reply:
Yeah I have that one on the radar to fix in Forms properly then as well. I think most people will set both of these anyway so that's why we might've gotten lucky here and not too much is broken here.